gridcapa-swe-runner-app/src/main/java/com/farao_community/farao/swe/runner/app/services/CgmesExportService.java

```diff
@@ -91,10 +90,23 @@ Map<String, ByteArrayOutputStream> generateCgmesFile(Network mergedNetwork, SweD
     Map<String, ByteArrayOutputStream> createAllSshFiles(Network mergedNetwork, SweData sweData) throws IOException {
         LOGGER.info("Building SSH files");
         Map<String, ByteArrayOutputStream> mapSshFiles = new HashMap<>();
-        Map<String, String> networkIdsByCountry = sweData.getMergedNetworkData().getSubnetworkIdByCountry();
-        mapSshFiles.putAll(createOneSsh(mergedNetwork.getSubnetwork(networkIdsByCountry.get("FR")), sweData, CgmesFileType.RTE_SSH));
-        mapSshFiles.putAll(createOneSsh(mergedNetwork.getSubnetwork(networkIdsByCountry.get("ES")), sweData, CgmesFileType.REE_SSH));
-        mapSshFiles.putAll(createOneSsh(mergedNetwork.getSubnetwork(networkIdsByCountry.get("PT")), sweData, CgmesFileType.REN_SSH));
+        Map<String, Network> subnetworksByCountry = new HashMap<>();
+        mergedNetwork.getSubnetworks().forEach(network -> {
+            if (network.getCountries().size() != 1) {
+                LOGGER.error("Subnetwork with id {} contains countries : {}, it will not be exported to SSH file", network.getNameOrId(), network.getCountries().toString());
+            } else {
+                String country = network.getCountries().stream().toList().get(0).name();
```

Comment:
Another slightly different way to do would be to use directly the `Country` object instead of a `String` containing the name of the country. This also allows us to use an `EnumMap` instead of a `hashMap`, which is supposed to be more efficient and compact when working with enum as keys. See proposal below:

```diff
-    private static final Map<String, String> TSO_BY_COUNTRY = Map.of("FR", "RTE", "ES", "REE", "PT", "REN");
+    private static final Map<Country, String> TSO_BY_COUNTRY = Map.of(Country.FR, "RTE", Country.ES, "REE", Country.PT, "REN");

    Map<String, ByteArrayOutputStream> createAllSshFiles(Network mergedNetwork, SweData sweData) throws IOException {
        [...]
-        Map<String, Network> subnetworksByCountry = new HashMap<>();
+        Map<Country, Network> subnetworksByCountry = new EnumMap<>(Country.class);
        [...]
            } else {
-                String country = network.getCountries().stream().toList().get(0).name();
+                Country country = network.getCountries().stream().toList().get(0);
                subnetworksByCountry.put(country, network);
            }
        });

-        for (Map.Entry<String, String> entry : TSO_BY_COUNTRY.entrySet()) {
-            String country = entry.getKey();
+        for (Map.Entry<Country, String> entry : TSO_BY_COUNTRY.entrySet()) {
+            Country country = entry.getKey();
        [...]
```